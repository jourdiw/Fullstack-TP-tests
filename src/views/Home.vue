<template>
  <v-jumbotron color="grey lighten-2">
    <v-container fill-height>
      <v-layout align-center>
        <v-flex>
          <h3 class="display-3">
            {{$t('home.title')}}
          </h3>
          <span class="subheading">
            {{$t('home.subheading')}}
          </span>
          <v-divider class="my-3"></v-divider>
          <div class="title mb-3">
            Ce tutoriel est accessible à titre de démo.
            Il est utilisé dans la
            <a href='https://makina-corpus.com/formations/formation-vuejs'>
              formation Vue.js
            </a>
            de Makina Corpus.
          </div>
          <router-link :to='link' class="nav-link">
            <v-btn large color="primary" class="mx-0">
              Découvrir l'application finale
              <v-icon>chevron_right</v-icon>
            </v-btn>
          </router-link>
        </v-flex>
      </v-layout>
    </v-container>
  </v-jumbotron>
</template>

<script>
export default {
  data: () => ({
    link: '/search?{"road": "52 rue Jacques Babinet", "city": "Toulouse", "zipCode": "31000", "complement": "Makina Corpus"}',
  }),
};
</script>


<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h1, h2 {
  font-weight: normal;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
