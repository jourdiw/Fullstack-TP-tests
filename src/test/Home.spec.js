import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';
import { shallowMount } from '@vue/test-utils';
import Home from '../views/Home.vue';

Vue.use(Vuetify);
Vue.use(VueRouter);

describe('Home.vue', () => {
  it('button is in a router link', () => {
    const wrapper = shallowMount(Home, {
      mocks: {
        $t: () => {},
      },
    });
    // expect(wrapper);
    // expect(wrapper.find('v-btn'));
    expect(wrapper.find('router-link'));
  });
  test('renders correctly', () => {
    const wrapper = shallowMount(Home, {
      mocks: {
        $t: () => {},
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
