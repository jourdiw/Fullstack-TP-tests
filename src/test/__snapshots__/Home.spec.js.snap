// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue renders correctly 1`] = `
<v-jumbotron-stub
  color="grey lighten-2"
  height="400px"
  tag="div"
>
  <v-container-stub
    fill-height=""
    tag="div"
  >
    <v-layout-stub
      align-center=""
      tag="div"
    >
      <v-flex-stub
        tag="div"
      >
        <h3
          class="display-3"
        >
          
          
        
        </h3>
         
        <span
          class="subheading"
        >
          
          
        
        </span>
         
        <v-divider-stub
          class="my-3"
        />
         
        <div
          class="title mb-3"
        >
          
          Ce tutoriel est accessible à titre de démo.
          Il est utilisé dans la
          
          <a
            href="https://makina-corpus.com/formations/formation-vuejs"
          >
            
            formation Vue.js
          
          </a>
          
          de Makina Corpus.
        
        </div>
         
        <router-link-stub
          class="nav-link"
          event="click"
          tag="a"
          to="/search?{\\"road\\": \\"52 rue Jacques Babinet\\", \\"city\\": \\"Toulouse\\", \\"zipCode\\": \\"31000\\", \\"complement\\": \\"Makina Corpus\\"}"
        >
          <v-btn-stub
            activeclass="v-btn--active"
            class="mx-0"
            color="primary"
            large="true"
            tag="button"
            type="button"
          >
            
            Découvrir l'application finale
            
            <v-icon-stub>
              chevron_right
            </v-icon-stub>
          </v-btn-stub>
        </router-link-stub>
      </v-flex-stub>
    </v-layout-stub>
  </v-container-stub>
</v-jumbotron-stub>
`;
